runner/components/io.siddhi.distribution.core/src/main/java/io/siddhi/distribution/core/persistence/S3PersistenceStore.java

```diff
@@ -0,0 +1,240 @@
+        try {
+            compressedSnapshot = CompressionUtil.compressGZIP(snapshot);
+        } catch (IOException e) {
+            log.error("Error occurred while trying to compress the snapshot. Failed to " +
+                    "persist revision: " + revision + " of Siddhi app: " + siddhiAppName);
+            return;
+        }
+        RequestBody requestBody = RequestBody.fromBytes(compressedSnapshot);
+        PutObjectResponse putObjectResponse =
+                s3Client.putObject(PutObjectRequest.builder().bucket(bucketName).key(revision)
+                        .build(), requestBody);
+        cleanOldRevisions();
+        log.debug("object has been uploaded to the bucket successfully, ETag: " + putObjectResponse.eTag());
+    }
+
+    @Override
+    public void setProperties(Map properties) {
+        Map configurationMap = (Map) properties.get(PersistenceConstants.STATE_PERSISTENCE_CONFIGS);
+        Object numberOfRevisionsObject = properties.get(PersistenceConstants.STATE_PERSISTENCE_REVISIONS_TO_KEEP);
+        String regionId;
+
+        if (!(numberOfRevisionsObject instanceof Integer)) { //it also check if it's Null or not
+            numberOfRevisionsToSave = PersistenceConstants.DEFAULT_REVISION_NO;
+            log.warn("Number of revisions to keep is not set or invalid. Default value will be used.");
+
+        } else {
+            numberOfRevisionsToSave = Integer.parseInt(String.valueOf(numberOfRevisionsObject));
+        }
+
+        if (configurationMap != null) {
+            Object regionObject = configurationMap.get(PersistenceConstants.REGION);
+            if (!(regionObject instanceof String)) {
+                regionId = PersistenceConstants.DEFAULT_REGION_ID;
+                log.info("No region id provided, Hence setting the region default region");
+            } else {
+                regionId = String.valueOf(regionObject);
+            }
+            Region region = Region.of(regionId.toLowerCase());
+            SdkHttpClient httpClient = ApacheHttpClient.builder().build();
+            S3ClientBuilder clientBuilder = S3Client.builder()
+                    .region(region);
+            AwsCredentialsProvider credentialsProvider = getCredentialProvider(configurationMap);
+            if (credentialsProvider != null) {
+                clientBuilder.credentialsProvider(credentialsProvider);
+            }
+            s3Client = clientBuilder.httpClient(httpClient).build();
+            List<Bucket> buckets;
+            try {
+                buckets = s3Client.listBuckets(ListBucketsRequest.builder().build()).buckets();
+            } catch (SdkClientException e) {
+                throw new S3PersistenceStoreException("Invalid region id provided, given region id: '" +
+                        regionId + "'.", e);
+            } catch (S3Exception e) {
+                throw new S3PersistenceStoreException("Invalid credential provided, check your user credential class" +
```

Comment:
How can we say that S3Exception throws only due to wrong credentials ?

Reply:
`buckets = s3Client.listBuckets(ListBucketsRequest.builder().build()).buckets();` will throw exception due to 2 reasons, 
1. Invalid region id
2. Invalid credential,

When the user gives the wrong credentials, S3Client will throw S3Exception with the following message
 `The AWS Access Key Id you provided does not exist in our records`, 
There's no other way to detect if the user has given the wrong credential or not.